index.js

```diff
@@ -63,6 +61,38 @@ module.exports.extra = function (pkg, deps, options) {
   return missing
 }
 
+function getDetective (name) {
+  try {
+    return name
+      ? (typeof name === 'string' ? require(name) : name)
+      : require('detective')
+  } catch (e) {}
+}
+
+function noopDetective () {
+  return []
+}
+
+function getExtensions (extensions, detective) {
+  var result = {
+    '.js': getDetective(detective),
+    '.node': noopDetective,
+    '.json': noopDetective
+  }
+
+  if (Array.isArray(extensions)) {
+    extensions.forEach(function (extension) {
+      result[extension] = getDetective(detective)
```

Comment:
I'll probably just leave this as-is. The `require` call should be cached, but I'd prefer to make the `getDetective` function cache instead of having to handle it multiple places.